### PR TITLE
feat: add grub

### PR DESCRIPTION
**Pull request by andrewrynhard**

### CODE REVIEW

File: grub/pkg.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                    --sysconfdir=/etc \
+                    --disable-efiemu \
+                    --disable-werror
+    build:
+      - |
+        make -j $(nproc)
+    install:
+      - |
+        make install DESTDIR=/rootfs
+        mkdir -p /rootfs/usr/share/bash-completion/completions
```
Comment (smira) [inline grub/pkg.yaml:28]: do we really need bash completion?

Comment (andrewrynhard) [reply grub/pkg.yaml:28]: Not in Talos. But this package should be able to stand on its' own. I envision that other distros might be able to help us maintain these packages in the future. Moving them to `/usr/share` ensures that they get removed by the Talos build: https://github.com/talos-systems/talos/blob/master/hack/cleanup.sh#L31.

**Pull request merged**